cpp/src/groupby/groupby.cu

```diff
@@ -147,7 +145,7 @@ groupby_groups groupby::groups(table_view values, rmm::mr::device_memory_resourc
 
   std::unique_ptr<table> group_values{nullptr};
   if (values.num_columns()) {
-    group_values = cudf::experimental::detail::gather(values, sort_order);
+    group_values = cudf::experimental::detail::gather(values, helper().key_sort_order());
```

Comment:
this won't work without a slice of key_sort_order

Reply:
Yup

Reply:
`key_sort_order` now returns a slice